### ISSUE TITLE
Search Bar

### ISSUE THREAD

**Issue by rayman89** [mbnuqw/sidebery#466]
Could you please add a search bar to search for tabs/bookmarks and quickly switch to them/open them? Thank you

**Comment by boldcompany** [comment]
Please do add this feature.

**Comment by emvaized** [comment]
Just saying — typing search query in regular FIrefox addressbar will also bring suggestions from opened tabs, with the ability to move to them on selecting

**Comment by rayman89** [comment]
That only works to jump to a tab, if you have a bunch of tabs and want to filter some that have something in common you can't do that. Also at least for me that firefox feature is full of flaws since if you have it enabled and type something to search and quickly press enter or if you want to open another tab from the same website but different subdomain it will jump you to an already open tab instead of opening the search engine.

**Comment by emvaized** [comment]
Okay, from this point of view it makes sense. So that, for example, you could type website domain in searchfield, and see all open tabs which match it by url or title.

**Comment by boldcompany** [comment]
Just checking to see if this feature is up for active consideration.

**Comment by emvaized** [comment]
@boldcompany This feature is already implemented in `v5`

**Comment by decard123** [comment]
> Just saying — typing search query in regular FIrefox addressbar will also bring suggestions from opened tabs, with the ability to move to them on select. So duplicating this feature in Sidebery seems a waste of time

1. most importantly, i don't think that searches Bookmarks, and certainly not just Bookmarks

2. i wasn't really aware that FF search seweaches tabs; but maybe it does. indeed it woudlk be nice if FF had a native search whgere you COULD select Web, Tabs, Bookmarks, Computer

**Comment by decard123** [comment]
> @boldcompany This feature is already implemented in `v5`

hmmm

1. it was a bit of search to try to find (possible) installed version. there is NO About tag on local version. all i could find is right down at the bottom Help in browser (not locally) in the fine print was version of FF and version of "Add-on: 4.10.2" which i only GUESSING means the version of SB.

2. also i see NO Update function locally OR in the browser help.
so not so helpful

**Comment by albino1** [comment]
FWIW, you can search Tabs with `%` in the URL bar and you can search Bookmarks with `*` in the URL bar.  So:

`% title of a tab`

`* name of a bookmark`

There's various other options listed here under `Changing results on the fly`:

https://support.mozilla.org/en-US/kb/address-bar-autocomplete-firefox

**Comment by decard123** [comment]
> FWIW, you can search Tabs with `%` in the URL bar and you can search Bookmarks with `*` in the URL bar. So:
> 
> `% title of a tab`
> 
> `* name of a bookmark`
> 
> There's various other options listed here under `Changing results on the fly`:
> 
> https://support.mozilla.org/en-US/kb/address-bar-autocomplete-firefox

excellent. thanks so much.

**Comment by decard123** [comment]
... so in that case it shoud lbe as easy as pie to embed those strings into Sidebnery as search options

btw, so can you add those parameters as shorts in FF Toolbar or sim?

**Comment by decard123** [comment]
and on 2nd issue, just WHERE is an Update function?

**Comment by albino1** [comment]
v5 is still in Beta and doesn't have a Mozilla Addons page, so to switch from v4 to v5 you have to install it manually by clicking the .xpi on the release page:

https://github.com/mbnuqw/sidebery/releases

This is the current version:

https://github.com/mbnuqw/sidebery/releases/tag/v5.0.0b27

**Comment by efermi** [comment]
I've tried the search (which I think should be called filter) in v5 beta: it takes a couple of *minutes* to filter through ~1400 tabs and the UI is locked the whole time. For comparison, on another machine with the TreeTabs addon it takes less than a second to search through ~2800 tabs.

**Comment by decard123** [comment]
quick question.
i looked at TreeTabs ages ago, but decided at the time that it was not what i wanted to use (at the moment, btw, i use OneTab and TabManager Plus very successfully (if only OneTab would add the obvious missing functions!)

so are TreeTab and Sideberry functionally analogous?

**Comment by efermi** [comment]
> so are TreeTab and Sideberry functionally analogous?

They are, at least for my purposes. The problem with TreeTabs is that it's been abandoned for years.

**Comment by decard123** [comment]
i was looking for a tool that would help me easily organise my 5,000+ Bookmarks & Bookmark Folders.

Plain old FF Bookmark Manager does that in a very limited slow way, but i was looking for something more easy/functional to use
.

**Comment by decard123** [comment]
for Tabs, my combination of OneTab and Tab Manager Plus work very well for me

**Comment by decard123** [comment]
i just wish they would support OneTab and implement the few obvious missing useful functions/enhancements, eg being able to move Tabs and blocks of Tabs much faster and more easily

**Comment by mbnuqw** [comment]
Available from v5.0.0. For global search follow #1064

**Comment by DRYN07** [comment]
> > FWIW, you can search Tabs with `%` in the URL bar and you can search Bookmarks with `*` in the URL bar. So:
> > `% title of a tab`
> > `* name of a bookmark`
> > There's various other options listed here under `Changing results on the fly`:
> > https://support.mozilla.org/en-US/kb/address-bar-autocomplete-firefox
> 
> excellent. thanks so much.

If I change the tab title, % does not work anymore on new titles. Do you have any solution for this, by any chance?

**Comment by albino1** [comment]
> If I change the tab title, % does not work anymore on new titles. Do you have any solution for this, by any chance?

You're right, edited titles don't work even with the Sidebery search.  You should create a new feature request for that as this one has already been implemented.
